packages/expo-modules-autolinking/src/index.ts

```diff
@@ -155,10 +165,22 @@ module.exports = async function (args: string[]) {
     const modules = await resolveModulesAsync(results, options);
     const extraDependencies = await resolveExtraBuildDependenciesAsync(options);
 
+    const coreFeatures = [
+      ...modules.reduce<Set<string>>((acc, module) => {
+        if (hasCoreFeatures(module)) {
+          const features = module.coreFeatures ?? [];
+          return new Set([...acc, ...features]);
```

Comment:
```suggestion
          for (const feature of features) {
            acc.add(feature);
          }
```

nit: not creating a new Set in each iteration.

also not sure if we want to check `typeof feature === 'string'` in case people provide an invalid coreFeatures in their expo-module.config.json
```json
{
  "coreFeatures": [{ "foo": "foo" }]
}
```

Reply:
We are currently not validating the types of other fields, and we should consider addressing this issue. However, we lack a good way to surface these problems to the user. We cannot display errors from the JS code because our native tools process the output. 

Let's discuss that problem and somehow address it. I'll create a linear task to not forget about it.

Reply:
that's right we had the same problem on other config fields. we can also improve that when that's an annoying problem.